src/RA_Defs.h

```diff
@@ -12,28 +14,63 @@
 #include <deque>
 #include <map>
 
+
 #ifndef RA_EXPORTS
 
 //	Version Information is integrated into tags
 
 #else
 
+
+// Maybe an extra check just in-case
+
+#if _HAS_CXX17
+#define _DEPRECATED          [[deprecated]]
+#define _DEPRECATEDR(reason) [[deprecated(reason)]]
+#define _FALLTHROUGH         [[fallthrough]]//; you need ';' at the end
+#define _NODISCARD           [[nodiscard]]
+#define _NORETURN            [[noreturn]]
+#define _UNUSED              [[maybe_unused]]
+
+// Disables the use of const_casts, if you get an error, it's not a literal
+// type. You could use it on functions but they will need a deduction guide
+// That would probably be better with a forwarding function
+
+
+
+
+#endif // _HAS_CXX17
 //NB. These must NOT be accessible from the emulator!
 //#define RA_INTEGRATION_VERSION	"0.053"
 
+//	RA-Only
+#define RAPIDJSON_HAS_STDSTRING 1
+// This shit man...
+#pragma warning(push, 1)
+// This is not needed the most recent version
+#define _SILENCE_ALL_CXX17_DEPRECATION_WARNINGS
 //	RA-Only
 #include "rapidjson/include/rapidjson/document.h"
 #include "rapidjson/include/rapidjson/reader.h"
 #include "rapidjson/include/rapidjson/writer.h"
 #include "rapidjson/include/rapidjson/filestream.h"
 #include "rapidjson/include/rapidjson/stringbuffer.h"
 #include "rapidjson/include/rapidjson/error/en.h"
+#pragma warning(pop)
 using namespace rapidjson;
 extern GetParseErrorFunc GetJSONParseErrorStr;
 
-#endif	//RA_EXPORTS
 
 
+using namespace std::string_literals;
+//using namespace std::chrono_literals; we could use this later
+namespace ra {}
+using namespace ra;
+
+#endif	//RA_EXPORTS
+#define _CONSTANT_VAR inline constexpr auto
+#define _CONSTANT     inline constexpr
```

Comment:
more seemingly unnecessary #defines

weren't you arguing against macros recently?

Reply:
I was arguing against macros for constants, keywords aren't used in execution, they're just placeholders.